x-pack/plugins/observability_solution/logs_explorer/public/customizations/types.ts

```diff
@@ -25,8 +26,15 @@ export interface LogsExplorerFlyoutContentProps {
   doc: LogDocument;
 }
 
+export type OnDiscoverNavigationHandler = (context: LogsExplorerControllerContext) => void;
+
+export interface LogsExplorerCustomizationEvents {
+  onDiscoverNavigation?: OnDiscoverNavigationHandler;
```

Comment:
qq: Looking from a Top Level, my understanding of this customisation is - 
What to do when user clicks on a Non Logs Type Dataset. 
- The user can chose to go to Discover
- The user can chose not to do anything, (in which case we disable the click)

What if user wants to do something completely different, arbitrarily thinking, they may chose to open a Popup,  or send use to Logs VIew in APM (worst case) 

If this a customisation, which decides what to do on click of a Dataset, why are we tying the name with `Discover`. The actual action could have nothing to do with Discover.

Should we not simply call it `onCustomNavigation` ?

Reply:
Agree, reading this again and thinking what this action represents, is not tied to Discover and not to navigation as well, what we want to react to is the case of an unknown data view beign selected/restored, I'll rename this to `onUknownDataViewSelection` to narrow it to its specific use case, thanks for the hint!